src/model/graph.ts

```diff
@@ -101,5 +103,66 @@ export class Graph implements GraphInterface {
     }
   }
 
+  toProtobuf(): protobuf.Graph {
+    let edges: { [key: string]: protobuf.Edge } = {};
+    Object.entries(this.edges).forEach(
+      (entry) => (edges[entry[0]] = entry[1].toProtobuf())
+    );
+    let nodes: { [key: string]: protobuf.Node } = {};
+    Object.entries(this.nodes).forEach(
```

Comment:
Instead of assigning a variable and then using `forEach` to fill it, you can also directly use `Object.entries(...).map(...)`. This should be the preferred style